### PR TITLE
External Media: Add account disconnect button

### LINKED ISSUE #16560: (8P) Google Photos: add account disconnect link in the media browser
Currently the only way to disconnect a Google Photos account from the external media feature is to go to My Site > Tools > Marketing on WordPress.com. This isn't the most obvious thing to do.

The Google Photos team have requested a disconnect link to be added somewhere in the media browser itself. This could either be in the modal title bar, or in the photo filters toolbar.

image

Note that there are UX guidelines around account disconnection:

https://developers.google.com/photos/library/guides/ux-guidelines

### PR DESCRIPTION
**Pull request by obenland**
Fixes #16560

#### Changes proposed in this Pull Request:

* Adds an `account` property to media state
* Expands the modal header with account information and an account disconnect button for Google Photos
* Adds an endpoint to the external media API that deletes a connection (limited to Google Photos).

#### Does this pull request change what data or activity we track or use?

Yes, tracks the service that is being disconnected.

#### Testing instructions:

* Apply D47813-code to your sandbox
* Sandbox public-api
* In the Editor, insert an image block and select Google Photos
* Connect to your Google Photos account.
* The account information should appear at the top right corner.
* Click on the disconnect button.
* After successful removal of the connection you should be redirected to the authentication screen.

#### Proposed changelog entry for your changes:

* External Media: Added a way to disconnect Google Photos accounts from within the media modal.

#### Screenshots:

Mobile:
image

Dektop:
image

### CODE REVIEW

**Comment by obenland** [comment]
> If I disconnect in wp-admin, will this break the connection for all other WordPress.com and / or Jetpack sites connected to my WordPress.com account that may rely on that connection? If so, maybe we should update our wording accordingly?

Is there a specific feature that you're thinking of that uses that connection and that I could test with?
Connection tokens seem to be user ID and service-specific, so all features that use that user's Google Photos connection will also lose it.

Do you have a suggestion for a more accurate wording?

**Comment by johngodley** [comment]
Personally the wording seems fine - it's clear it's for the user, and this matches Calypso. If someone does accidentally disconnect then it is easy to reconnect, and they are prompted to do so the next time they try to access the feature on any site.

**Comment by obenland** [comment]
@jeherve Thanks for the heads up. I added a fallback in case `image` and `name` might not be set at any point in the future. Unless I missed something, that shouldn't happen with D47813-code applied and public-api sandboxed.

File: extensions/shared/external-media/sources/google-photos/google-photos-account.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	const { image, name } = account || {};
+
+	return (
+		<div className="jetpack-external-media-header__account">
+			<div className="jetpack-external-media-header__account-info">
+				{ image && (
+					<img
+						className="jetpack-external-media-header__account-image"
+						src={ image }
+						alt=""
```
[the remark this answers is not in the document]
Comment (obenland) [reply extensions/shared/external-media/sources/google-photos/google-photos-account.js:16]: `Profile picture of {name}`?

[the remark this answers is not in the document]
Comment (jeryj) [reply extensions/shared/external-media/sources/google-photos/google-photos-account.js:16]: I mean if there is alt text passed along with the image, use it. `Profile picture of {name}` is a pretty good idea though... I think that's a good solution as that's the purpose of the image.

**Comment by obenland** [comment]
>  @jeherve removed the [Status] Ready to Merge label

Just to clarify, this shouldn't be merged just yet?

**Comment by jeherve** [comment]
> Just to clarify, this shouldn't be merged just yet?

I removed the label since I saw @jeryj had some feedback. I'll leave it to him to decide whether this is a blocker or not. 👍

**Comment by jeryj** [comment]
I'm fine with merging and taking care of the focus loss in a follow-up PR.

**Comment by obenland** [comment]
I _will_ wait with merging until D47813-code was approved, though :)

**Comment by obenland** [comment]
Deployed in r212124-wpcom

**Pull request merged**